streams/src/main/java/org/apache/kafka/streams/kstream/Window.java

```diff
@@ -56,9 +56,7 @@ public long end() {
      * @param other  another window
      * @return       {@code true} if {@code other} overlaps with this window&mdash;{@code false} otherwise
      */
-    public boolean overlap(Window other) {
-        return this.start() < other.end() || other.start() < this.end();
-    }
+    public abstract boolean overlap(Window other);
```

Comment:
Yes, it should be `abstract` because `Window` does not know if boundaries will be inclusive or exclusive. And implementation for `SessionWindow` and `TimeWindow` are different IMHO -- or do I miss anything -- at least, both should be different because `SessionWindow` does have *inclusive end time* while `TimeWindow` has *exclusive end time*.